packages/govuk-frontend-review/src/common/middleware/feature-flags.mjs

```diff
@@ -0,0 +1,57 @@
+import express from 'express'
+
+const router = express.Router()
+
+const COOKIE_NAME = 'use_recolour'
+
+/**
+ * Control cookie setting and unsetting when the setRecolour query is requested
+ *
+ * 3 possible states:
+ *
+ * 1. setRecolour == true: set the recolour cookie and redirect
+ * 2. setRecolour == 'all-flag-states': set the showAllFlagStates local variable
+ *    for the immediate view and don't redirect
+ * 3. setRecolour is present but doesn't satisfy the above: unset the recolour
+ *    cookie and redirect
+ */
+router.use((req, res, next) => {
+  const urlWithoutQueryString = req.url.split('?')[0]
+
+  if (req.method === 'GET' && 'setRecolour' in req.query) {
+    if (req.query.setRecolour === 'true') {
+      const maxAgeInDays = 28
+
+      res.cookie(COOKIE_NAME, 'yes', {
+        maxAge: maxAgeInDays * 24 * 60 * 60 * 1000,
+        httpOnly: true
+      })
+
+      return res.redirect(urlWithoutQueryString)
+    } else if (req.query.setRecolour === 'all-flag-states') {
+      res.locals.showAllFlagStates = true
```

Comment:
In practice this is more like a recolour override, since it doesn't set a cookie and would override what the cookie specifies since it would show both examples. Should this be in the other middleware and use the `recolourOverride` param?

Reply:
I concur it's a bit weird.

I structured it like that to account for the radio buttons-based UI but also to issolate the feature. `setRecolour` redirects after setting or unsetting the cookie, which disrupts the setting of this. I reckon the solution to that is to change how the redirects in this middleware happen so that they only remove the `setRecolour` param and not the whole query list. Then I can action this feedback.

I also wonder if it makes sense to separate the both states part of the UI into its own question like a checkbox.